src/test/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AFSimulatedData.java

```diff
@@ -0,0 +1,111 @@
+    private final AFData data;
+    private final int numSegments;
+
+    public AFSimulatedData(final double averageHetsPerSegment, final int numSegments,
+                           final double averageDepth, final double biasMean, final double biasRate, final double pOutlier) {
+
+        this.numSegments = numSegments;
+        final AFState.MinorFractions minorFractions = new AFState.MinorFractions();
+        final List<AllelicCount> alleleCounts = new ArrayList<>();
+        final List<IntRange> segmentRanges = new ArrayList<>();
+
+        final PoissonDistribution segmentLengthGenerator = new PoissonDistribution(averageHetsPerSegment);
+        final PoissonDistribution readDepthGenerator = new PoissonDistribution(averageDepth);
+        final UniformRealDistribution minorFractionGenerator = new UniformRealDistribution(0.0, 0.5);
+
+        //translate to ApacheCommons' parametrization of the gamma distribution
+        final double gammaShape = biasMean * biasRate;
+        final double gammaScale = 1 / biasRate;
+        final GammaDistribution biasGenerator = new GammaDistribution(gammaShape, gammaScale);
+        final Random rng = new Random();
+
+        int hetAtStartOfSegment = 0;
+        for (int segment = 0; segment < numSegments; segment++) {
+            final double minorFraction = minorFractionGenerator.sample();
+            minorFractions.add(minorFraction);
+
+            final int numHetsInSegment = Math.max(MIN_HETS_PER_SEGMENT, segmentLengthGenerator.sample());
+            final int hetAtEndOfSegment = hetAtStartOfSegment + numHetsInSegment;
+            segmentRanges.add(new IntRange(hetAtStartOfSegment, hetAtEndOfSegment));
+
+            for (int het = hetAtStartOfSegment; het < hetAtEndOfSegment; het++) {
+                final double bias = biasGenerator.sample();
+
+                //flip a coin to decide alt minor (alt fraction = minor fraction) or ref minor (alt fraction = 1 - minor fraction)
+                final double altFraction =  rng.nextDouble() < 0.5 ? minorFraction : 1 - minorFraction;
+
+                //the probability of an alt read is the alt fraction modified by the bias or, in the case of an outlier, random
+                final double pAlt = rng.nextDouble() < pOutlier ? rng.nextDouble() : altFraction / (altFraction + (1 - altFraction) * bias);
+
+                final int numReads = readDepthGenerator.sample();
+                final int numAltReads = new BinomialDistribution(numReads, pAlt).sample();
+                final int numRefReads = numReads - numAltReads;
+                alleleCounts.add(new AllelicCount(DUMMY_INTERVAL, numRefReads, numAltReads));
+            }
+            hetAtStartOfSegment = hetAtEndOfSegment;
+        }
+
+        data = new AFData(alleleCounts, segmentRanges);
+        trueState = new AFState(biasMean, biasRate, pOutlier, minorFractions);
+    };
+
+    public AFState getTrueState() { return trueState; }
+    public AFData getData() { return data; }
```

Comment:
see other comments; i think that this class should be able to return a full SegmentedModel